ports/hlslpp/hlslpp-config.cmake

```
@@ -0,0 +1,4 @@
+add_library(hlslpp INTERFACE)
+add_library(hlslpp::hlslpp ALIAS hlslpp)
+
+target_include_directories(hlslpp INTERFACE "${CMAKE_CURRENT_LIST_DIR}/../../include/hlslpp")

```

The missing piece was the `IMPORTED` keyword.
And as long as it isn't provided by upstream, it package name and target name shall be prefixed as `unofficial`.
```suggestion
add_library(unofficial::hlslpp::hlslpp INTERACE IMPORTED)
target_include_directories(unofficial::hlslpp::hlslpp INTERFACE "${CMAKE_CURRENT_LIST_DIR}/../../include/hlslpp")
```
installed as `share/unofficial-hlslpp/unofficial-hlslpp-config.cmake`.